### PR TITLE
chore: Update README with backward compat. information

### PR DESCRIPTION
**Pull request by hf**
Adds information to the README on guarantees for backward compatibility especially with regards to self-hosting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +39,132 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
...
+security updates.
+
+If you still insist on running it on your own, please make sure you setup a
+process to promptly update to the latest version. You can do that by following
+this repository, specifically the
+[Releases](https://github.com/supabase/gotrue/releases) and [Security
+Advisories](https://github.com/supabase/gotrue/security/advisories) sections.
+
+You should also subscribe to the
+[supabase-gotrue-announcements](https://groups.google.com/g/supabase-gotrue-users)
```
Comment (hf) [inline README.md:55]: Need to create the group before merging, ofc.

**Review by kangmingtay — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -31,6 +39,131 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
 3. `docker ps` should show 2 docker containers (`gotrue_postgresql` and `gotrue_gotrue`)
 4. That's it! Visit the [health checkendpoint](http://localhost:9999/health) to confirm that gotrue is running.
 
+## Running in production
+
+Running an authentication server in production is not an easy feat. We
+recommend using [Supabase Auth](https://supabase.com/auth) which gets regular
```
Comment (J0) [inline README.md:45]: Supabase Auth 🎉

File: README.md [hunk 3]
```diff
@@ -31,6 +39,131 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
 3. `docker ps` should show 2 docker containers (`gotrue_postgresql` and `gotrue_gotrue`)
 4. That's it! Visit the [health checkendpoint](http://localhost:9999/health) to confirm that gotrue is running.
```
Comment (J0) [inline README.md:40]: /nit know this isn't in the PR but should this be 
```suggestion
4. That's it! Visit the [health check endpoint](http://localhost:9999/health) to confirm that gotrue is running.
```

File: README.md [hunk 4]
```diff
@@ -31,6 +39,131 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
...
+changes.
+
+**Patch**
+
+Changes to the patch version guarantees backward compatibility with:
+
+- Database objects (tables, columns, indexes, functions).
+- REST API
+- JWT structure
+- Configuration
```
Comment (J0) [inline README.md:75]: Should we specify that this refers to the names of the configuration values which live in the [Environment Variable files](https://github.com/supabase/gotrue/blob/master/example.env)?

Imagine that we might want to change the properties of the logger down the road which are also part of the configuration atm

File: README.md [hunk 5]
```diff
@@ -31,6 +39,131 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
...
+comprehensive list of those features:
+
+1. Multi-tenancy via the `instances` table i.e. `GOTRUE_MULTI_INSTANCE_MODE`
+   configuration parameter.
+2. System user (zero UUID user).
+3. Super admin via the `is_super_admin` column.
+4. SAML sign-in provider via the `GOTRUE_SAML_ENABLED` configuration
+   parameter. (A different implementation for SAML may appear in the future
+   which will be supported.)
+5. Support for MySQL based databases. (Only Postgres is supported.)
```
Comment (J0) [inline README.md:146]: Hmm are we using any Postgres specific features which would completely block a user from self-hosting on MySQL?

If not we could consider stating that it's possible but not natively supported (e.g. we won't have the starter files and we won't be taking queries about self-hosting with MySQL but the dedicated user could self-host with MySQL if they added additional code). 

This defers the decision of tying GoTrue to Postgres and allows us to eventually come back and support other Databases if we want to. At the same time, I realise it is quite unlikely that we will want to support other DBs given that the rest of Supabase tooling is heavily dependent on Postgres.


```suggestion
5. Built-in support for MySQL based databases. (We primarily support Postgres)
```

**Review by J0 — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (darora) [reply README.md:146]: imo @hf 's wording is sane. If we want to add support for other DBs in the future that's fine, we can always come back and update this.

Comment (J0) [reply README.md:146]: Ok! Resolving

**Pull request merged**